src/lang/proc_macros/db.rs

```diff
@@ -107,25 +117,63 @@ pub fn get_derive_expansion(
             diagnostics: Default::default(),
             code_mappings: None,
         }
-    })
+    });
+
+    correct_code_mappings(&mut result.code_mappings, call_site);
+
+    result
 }
 
 pub fn get_inline_macros_expansion(
     db: &dyn ProcMacroGroup,
-    params: ExpandInlineMacroParams,
+    mut params: ExpandInlineMacroParams,
 ) -> ProcMacroResult {
-    db.get_stored_inline_macros_expansion(params.clone().into()).unwrap_or_else(|| {
-        // We can't return the original node because it will make us fall into infinite recursion.
-        let unit = "()".to_string();
+    let call_site = replace_call_site(&mut params.call_site);
 
-        if let Some(client) = db.proc_macro_server_status().ready() {
-            client.request_inline_macros(params);
-        }
+    let mut result =
+        db.get_stored_inline_macros_expansion(params.clone().into()).unwrap_or_else(|| {
+            // We can't return the original node because it will make us fall into infinite recursion.
+            let unit = "()".to_string();
 
-        ProcMacroResult {
-            token_stream: cairo_lang_macro_v1::TokenStream::new(unit),
-            diagnostics: Default::default(),
-            code_mappings: None,
+            if let Some(client) = db.proc_macro_server_status().ready() {
+                client.request_inline_macros(params);
+            }
+
+            ProcMacroResult {
+                token_stream: cairo_lang_macro_v1::TokenStream::new(unit),
+                diagnostics: Default::default(),
+                code_mappings: None,
+            }
+        });
+
+    correct_code_mappings(&mut result.code_mappings, call_site);
+
+    result
+}
+
+fn replace_call_site(call_site: &mut TextSpan) -> TextSpan {
+    let zero_based_span = TextSpan { start: 0, end: call_site.end - call_site.start };
+
+    std::mem::replace(call_site, zero_based_span)
+}
+
+fn correct_code_mappings(code_mappings: &mut Option<Vec<CodeMapping>>, call_site: TextSpan) {
```

Comment:
I am missing information what are you exactly doing here ("correct") and why we need it